internal/testrunner/runners/asset/tester.go

```diff
@@ -191,20 +211,67 @@ func (r *tester) TearDown(ctx context.Context) error {
 	return nil
 }
 
-func findActualAsset(actualAssets []packages.Asset, expectedAsset packages.Asset) bool {
+func findActualAsset(actualAssets []packages.Asset, savedObjects []common.MapStr, expectedAsset packages.Asset) bool {
 	for _, a := range actualAssets {
 		if a.Type == expectedAsset.Type && a.ID == expectedAsset.ID {
 			return true
 		}
 	}
 
+	if expectedAsset.Type == "tag" && expectedAsset.ID == "" {
+		// If we haven't found the asset, and it is a tag, it could be some of the shared
+		// tags defined in tags.yml, whose id can be unpredictable, so check by name.
+		if len(actualAssets) == 0 {
+			// If there are no assets, the tag may not be installed, so assume it would have been.
+			// TODO: More accurately we should check if any of the listed tags in `tags.yml` is present.
```

Comment:
This would be actually wrong in the package, and can require quite some changes to be supported as we would need to bring here the whole `tags.yml`. So if we find a package with this I would prefer to fix this in the package.